### PR TITLE
Image import - memory hog fix

### PR DESCRIPTION
**Pull request by shurkaxaa**
Existing version load image into memory when non-unified image used
(with separate tarball with manifest).
Added ability for stream based image upload without loading them to memory.
requests-toolbelt external package required for streaming multipart HTTP
image upload.

### CODE REVIEW

File: pylxd/models/image.py [hunk 1]
```diff
@@ -119,7 +122,17 @@ def create(
         if public:
             headers['X-LXD-Public'] = '1'
 
-        if metadata is not None:
+        if from_streams:
```
Comment (ChrisMacNaughton) [inline pylxd/models/image.py]: Is there a compelling reason that we'd ever want to _not_ stream the image? If not, I'd prefer that we not add a new option and just chunk all image uploads

Comment (shurkaxaa) [reply pylxd/models/image.py]: I wrote it in comment - the only reason if we want to have pylxd clients to work with old version too.
For example nova-lxd - if it will be switched to stream based image upload it will require NEW version of pyxld i.e it will not be able to wold with old version of pylxd. I do not see any versioning in pylxd code so client will not be able which version of pyxld is installed.
As I was not sure that pyxld and nova-lxd will be updated synchronously by packagers I assume that clients can fallback to non-stream behavior via introspection of create method.
In short it will allow client of pyxld library to work with both version of pyxld - fixed one and old one. If we are sure that pyxld will be ALWAYS updated as it's client we can remove this for sure and assume that streams always passed to create operation.

Comment (shurkaxaa) [reply pylxd/models/image.py]: May be it make sense at least to dynamically identify if streams are passed or content as before and act accordingly?
It will provide backward compatibility for 'old' clients which will pass content instead of streams. What do you think?

File: pylxd/tests/models/test_image.py [hunk 2]
```diff
@@ -110,7 +111,8 @@ def test_create_with_metadata(self):
         """An image with metadata is created."""
         fingerprint = hashlib.sha256(b'').hexdigest()
         a_image = models.Image.create(
-            self.client, b'', metadata=b'', public=True, wait=True)
+            self.client, StringIO(u''), metadata=StringIO(u''),
```
Comment (ChrisMacNaughton) [inline pylxd/tests/models/test_image.py]: Instead of editing this test, can you please add another test that verifies the new functionality while allowing us to maintain a test to ensure no future regressions in the API handling?

**Comment by shurkaxaa** [comment]
Old functionality no longer available if I will keep old test it will fail. You ask me before to always assume that stream based upload used, so modified test no longer relevant as it does not use streams. If I will keep it as is it will fail.

**Comment by shurkaxaa** [comment]
'Old' test returned back as requested.
I've checked that requests_toolbelt multipart encoder
accept both types on input - binary string and file like object.
It means that backward compatibility with old clients guaranteed.
So old clients will still upload images as binary strings and new one can use streams.

**Comment by ChrisMacNaughton** [comment]
@ajkavanagh Thoughts?
